receiver/k8slogreceiver/README.md

```diff
@@ -0,0 +1,144 @@
+
+| Field         | Description                                                  |
+|---------------|--------------------------------------------------------------|
+| `annotations` | MapFilters that filters pods by annotations.                 |
+| `labels`      | MapFilters that filters pods by labels.                      |
+| `env`         | MapFilters that filters containers by environment variables. |
+| `containers`  | ValueFilters that filters containers by name.                |
+| `namespaces`  | ValueFilters that filters pods by namespace.                 |
+| `pods`        | ValueFilters that filters pods by name.                      |
+
+#### MapFilter
+
+A MapFilter can be used to filter pods by maps, such as annotations or labels. It has the following fields:
+
+| Field   | Description                                                                                                                                                                                                                                                                                                                                                                                                                               |
+|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `op`    | The operation to perform. Options are: <br> - "equals": (default) the value must be equal to the specified value. <br>- "not-equals": the value must not be equal to the specified value. <br> - "exists": the value must exist. <br> - "not-exists": the value must not exist. <br> - "matches": the value must match the specified regular expression. <br> - "not-matches": the value must not match the specified regular expression. |
+| `key`   | The key of the map.                                                                                                                                                                                                                                                                                                                                                                                                                       |
+| `value` | The value to match. Only used for "equals", "not-equals", "matches", and "not-matches" operations.                                                                                                                                                                                                                                                                                                                                        |
```

Comment:
Do we need this right away?

Note other components also have built their own filters, such as k8sobjects.

Reply:
This filter is similar to the `k8sattributes` processor. Many users who want to use this component may be familiar with k8sattributes since it was the previous solution for correlating k8s metadata to logs. This reduces their migration costs.